packages/@livestore/common/src/sync/ClientSessionSyncProcessor.ts

```diff
@@ -100,29 +103,29 @@ export const makeClientSessionSyncProcessor = ({
         }),
       )
     })
-
-    const mergeResult = SyncState.merge({
-      syncState: syncStateRef.current,
-      payload: { _tag: 'local-push', newEvents: encodedEventDefs },
-      isClientEvent,
-      isEqualEvent: LiveStoreEvent.isEqualEncoded,
-    })
+    yield* Effect.annotateCurrentSpan({ batchSize: encodedEventDefs.length })
+
+    const mergeResult = yield* Effect.sync(() =>
+      SyncState.merge({
+        syncState: syncStateRef.current,
+        payload: { _tag: 'local-push', newEvents: encodedEventDefs },
+        isClientEvent,
+        isEqualEvent: LiveStoreEvent.isEqualEncoded,
+      }),
+    )
 
     if (mergeResult._tag === 'unexpected-error') {
-      return shouldNeverHappen('Unexpected error in client-session-sync-processor', mergeResult.message)
+      return yield* Effect.die(new Error(`Unexpected error in client-session-sync-processor: ${mergeResult.message}`))
     }
 
-    span.addEvent('local-push', {
-      batchSize: encodedEventDefs.length,
-      mergeResult: TRACE_VERBOSE ? JSON.stringify(mergeResult) : undefined,
-    })
```

Comment:
This OTel event is now covered by the `client-session-sync-processor:push` span.